utils/api_util.py

```diff
@@ -0,0 +1,111 @@
+"""Ingests historical CSV data into DB."""
+
+import json
+import os
+from datetime import datetime
+
+import daiquiri
+import pandas as pd
+import requests
+
+from utils import cloud_constants as cc
+from utils.storage_utils import get_file_prefix, save_data_to_csv
+
+log = daiquiri.getLogger(__name__)
+
+INSERT_API_PATH = "/api/v1/pcve"
+API_FULL_PATH = "{host}{api}".format(host=cc.OSA_API_SERVER_URL, api=INSERT_API_PATH)
+
+failed_to_insert = []
+
+
+def _report_failures(df: pd.DataFrame, triage_subdir: str, s3_upload: bool, ecosystem: str):
+    """Save failed record."""
+    if len(failed_to_insert) == 0:
+        log.info("Successfully ingested all records")
+    else:
+        failed_list_str = ",".join(failed_to_insert)
+        log.error("Failed to insert {count} data : {data}".format(count=len(failed_to_insert), data=failed_list_str))
+
+        failed_data = df[df['url'].isin(failed_to_insert)]
+        save_data_to_csv(failed_data, s3_upload, cc.FAILED_TO_INSERT, triage_subdir, ecosystem, cc.PROBABLE_CVES)
+
+        log.info("Failed data saved successfully.")
+
+
+def _insert_df(df: pd.DataFrame, url: str):
+    """Call API server and insert data."""
+    objs = df.to_dict(orient='records')
+    for obj in objs:
```

Comment:
Instead of sending this one at a time can the API handle chunks of these, say 10-15 at a time? /cc @arajkumar

Reply:
Yes we can do same like we did for osa-api bulk insert using asyncio. Let me make that change so both place it will be same.

Reply:
done